### ISSUE TITLE
Windows Guide

### ISSUE THREAD

**Issue by hartontw** [jakkn/nwn-devbase#26]
Maybe I have not searched correctly but I don't know how to follow.
nwn-build don't work.

# Windows

### [Install Chocolatey](https://chocolatey.org/install#installing-chocolatey)

Install with cmd.exe
```
@"%SystemRoot%\System32\WindowsPowerShell\v1.0\powershell.exe" -NoProfile -InputFormat None -ExecutionPolicy Bypass -Command "iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))" && SET "PATH=%PATH%;%ALLUSERSPROFILE%\chocolatey\bin"
```

Install with PowerShell.exe
```
Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
```

Install from PowerShell v3+
```
Set-ExecutionPolicy Bypass -Scope Process -Force; iwr https://chocolatey.org/install.ps1 -UseBasicParsing | iex
```

### [Install git](https://git-scm.com/)

[Download](https://git-scm.com/download/win)

or
```
choco install git
```

### [Install Ruby](https://rubyinstaller.org/downloads/)
```
choco install ruby
```

### [Install nim](https://github.com/dom96/choosenim)

Download nim:
- [Self-extracting archive](https://github.com/dom96/choosenim/releases/download/v0.3.2/choosenim-0.3.2_windows_i386.exe)
or 
- [Zip archive](https://github.com/dom96/choosenim/releases/download/v0.3.2/choosenim-0.3.2_windows_i386.zip), extract and run the ```runme.bat```

Add nim to [PATH](https://www.architectryan.com/2018/03/17/add-to-the-path-on-windows-10/):
```
C:\Users\YOUR_USER\.nimble\bin
```
or where your nim is installed (you can check it when installs).

### [Install neverwinter_utils.nim](https://github.com/niv/neverwinter.nim)
```
nimble install neverwinter
```

### [Install nwnsc](https://neverwintervault.org/project/nwnee/other/tool/nwnsc-nwn-enhanced-edition-script-compiler)
1. [Download](https://neverwintervault.org/sites/all/modules/pubdlcnt/pubdlcnt.php?file=https://neverwintervault.org/sites/neverwintervault.org/files/project/29016/files/nwnsc-windows-1.0.0.zip&nid=29016)
2. Extract in some folder under your watch
3. Add nwnsc.exe folder to [PATH](https://www.architectryan.com/2018/03/17/add-to-the-path-on-windows-10/)

### [Install nwn-devbase](https://github.com/jakkn/nwn-devbase)
```
git clone https://github.com/jakkn/nwn-devbase.git
cd nwn-devbase
gem install bundler
bundle install
```

# ...

### [Install Docker](https://hub.docker.com/editions/community/docker-ce-desktop-windows)
1. __Sign in__ or __Sign Up__
2. [Download](https://download.docker.com/win/stable/Docker%20for%20Windows%20Installer.exe)
3. Install
4. Close and log out
5. Log in
6. Enable Hyper-V feature
7. Restart

Shows errors:
```Powershell
docker run --rm -it -v "$(pwd)/home/devbase/build" jakkn/nwn-devbase pack
```
Don't:
```Powershell
docker run --rm -it -v "$(pwd):/home/devbase/build" jakkn/nwn-devbase pack
```

**Comment by jakkn** [comment]
Hi, I struggle to understand your problem. Can you please explain again and be more specific? Also, you have listed both native and docker install steps. Only one is necessary. Which one are you using? If you have errors please attach them.

**Comment by hartontw** [comment]
Hi,

Sorry, I have not expressed myself correctly.

I'm trying to write guides for every OS from the beginning to the first unpack and pack.

I have not experience with docker so I've tried to start with natively workaround. The docker part at the bottom is myself pasting too much by mistake. When I have more experience I will start with the guides for Docker.

Command ```nwn-build``` doesn't work for me. Probably I miss something due to my lack of experience, but I don't know what.

**Comment by jakkn** [comment]
Ah I see! That's great to hear, PRs are most welcome :) I've felt for a long time that the README needs an overhaul to reduce the amount of text.

You're probably just missing the symlink described under [Paths](https://github.com/jakkn/nwn-devbase#paths). I couldn't figure out how to add `nwn-build` to PATH on Windows so it became a manual step instead, unfortunately.

Regarding install instructions, I have deliberately avoided including instructions for dependencies because they do not belong here. The command to install chocolatey for instance is not something I want in the README for several reasons, the most important of which are trust and maintenance. Please only include references to official pages where the install instructions are hosted and no copies of commands that install software. The only exception is package manager install commands, like `choco install foo`, `apt install bar`, etc.

Looking over the content now I'm starting to think that it may be better to split the instructions in separate files for the different platforms, something like INSTALL.LIN.md, INSTALL.WIN.md, INSTALL.OSX.md. What do you think?

**Comment by hartontw** [comment]
You're right about excluding instructions for dependencies because these are subject to change.

I understand what I've missing and now I can finish now Windows Guide Natively workaround. When I have it I will make a pull request with the name you suggested (```INSTALL.WIN.md```).

**Comment by devpitOne** [comment]
Hi, I have the docker version up and running on Windows. I haven't aliased the container to nwn-build yet but when I do I'll let you know.
